src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -1256,11 +1265,58 @@ void BluezObjectsCleanup(BluezEndpoint * apEndpoint)
     EndpointCleanup(apEndpoint);
 }
 
+#if !CHIP_BYPASS_ADDITIONAL_DATA_ADVERTISING
+static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * characteristic)
+{
+    if (characteristic == nullptr)
+    {
+        return;
+    }
+
+    // Construct the TLV for the additional data
+    GVariant * cValue = nullptr;
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+    TLVWriter writer;
+    TLVWriter innerWriter;
+    chip::System::PacketBufferHandle bufferHandle = chip::System::PacketBuffer::New();
+    chip::System::PacketBuffer * buffer           = bufferHandle.Get_ForNow();
+
+    writer.Init(buffer);
+
+    err = writer.OpenContainer(ProfileTag(kProtocol_ServiceProvisioning, kTag_AdditionalDataExensionDescriptor), kTLVType_Structure,
```

Comment:
Something here should set `writer.ImplicitProfileId` to get implicit profile tag encoding like the spec says.

Also, the tag (kProtocol_ServiceProvisioning, 0) is already defined to be `kTag_QRCodeExensionDescriptor` as far as I can tell.  So this needs to use either a different profile or a different tag value (and the spec should be fixed accordingly).

Reply:
i don't see a problem with using the same tag/profile here as well, those are two different payloads that will never be combined.

Reply:
The whole point of profiles and tags in TLV is that a fully qualified tag uniquely identifies what you are working with.

If this payload is always known to be "additional data", we should just say that (based on the context it appears in) and use an anonymous tag for the outermost struct; that will save a few bytes on the wire too.  The point of using a profile-based tag for the outermost container is if there is not enough context to tell what the thing is, so the tag is needed to disambiguate.

Reply:
I got it, i create an issue #4259 , meanwhile, i will use anonymous tag for the outer most tag because it seems like the most suitable solution for now.